lib/shelley/bench/Restore.hs

```diff
@@ -316,24 +319,160 @@ cardanoRestoreBench tr c socketFile = do
     networkDescription :: forall n. (NetworkDiscriminantVal n) => Proxy n -> Text
     networkDescription _ = networkDiscriminantVal @n
 
+
+seed :: SomeMnemonic
+seed = SomeMnemonic . unsafeMkMnemonic @15 $ T.words
+    "involve key curtain arrest fortune custom lens marine before \
+    \material wheel glide cause weapon wrap"
+
 {-------------------------------------------------------------------------------
                                   Benchmarks
 -------------------------------------------------------------------------------}
 
-data BenchResults = BenchResults
+data SomeBenchmarkResults where
+    SomeBenchmarkResults
+        :: forall result. Buildable result
+        => result
+        -> SomeBenchmarkResults
+
+instance Buildable SomeBenchmarkResults where
+    build (SomeBenchmarkResults results) = build results
+
+data BenchRndResults s = BenchRndResults
     { benchName :: Text
-    , restorationTime :: Time
-    , listingAddressesTime :: Time
-    , estimatingFeesTime :: Time
-    , utxoStatistics :: UTxOStatistics
-    } deriving (Show, Generic)
+    , restoreTime :: Time
+    , readWalletTime :: Time
+    , listAddressesTime :: Time
+    , importOneAddressTime :: Time
+    , importManyAddressesTime :: Time
+    , estimateFeesTime :: Time
+    , utxoStatistics :: s
```

Comment:
I don't see: why do you need to parametrise over `s`?

Reply:
Because I didn't want to define a Buildable instance for ApiUtxoStatistics.